File: Facultate/Anul 2/Sem 2/MPP/Laburi/Edi/JavaClient C# Server/Server/src/main/java/utils/ServiceFactory.java
package utils;

import repos.EmployeeDBRepo;
import repos.HibernateReservationsRepo;
import repos.HibernateRidesRepo;
import services.AuthService;
import services.MainService;


public class ServiceFactory {

     private static ServiceFactory instance;

     private ServiceFactory() {

     }

     public static ServiceFactory getInstance() {
          if (instance == null) {
               instance = new ServiceFactory();
          }
          return instance;
     }

     public AuthService getAuthService() {
         return new AuthService(new EmployeeDBRepo(Props.getProperties()));
     }

     public MainService getMainService() {
            return new MainService(
                    new EmployeeDBRepo(Props.getProperties()),
                    new HibernateRidesRepo(),
                    new HibernateReservationsRepo()
            );
     }

}

File: Facultate/Anul 2/Sem 2/MPP/Laburi/Edi/CSharpServer/Server/factories/ServiceFactory.cs
namespace Server.factories;

using repos;
using services;
using utils;
using Shared.services;

public class ServiceFactory
{
    private static ServiceFactory? _instance;
    private readonly DbUtils _dbUtils;
    
    public static ServiceFactory Instance => _instance ??= new ServiceFactory();
    
    private ServiceFactory()
    {
        _dbUtils = new DbUtils("C:\\Users\\Deny\\Documents\\GitHub\\mpp-proiect-csharp-Manu-Ioan-Daniel\\CSharpLab_3\\Client\\db.properties");
    }

    public AuthService GetAuthService() =>
        new(new EmployeeDbRepo(_dbUtils));

    public IMainService GetMainService()
    {
        return new MainService(
            new EmployeeDbRepo(_dbUtils),
            new RidesDbRepo(_dbUtils),
            new ReservationsDbRepo(_dbUtils)
        );
    }

}